src/pointings_search/pointing_table.py

```diff
@@ -0,0 +1,303 @@
+        for key in alt_names.keys():
+            self._check_and_rename_column(key, alt_names[key], required=False)
+
+    def append_sun_pos(self, precise=False, recompute=False):
+        """Compute an approximate position of the sun (relative to Earth) at
+        each time and append this the table. Caches the result within the table
+        so that this computation only needs to be performed once.
+
+        Parameters
+        ----------
+        precise : `bool`
+            Use the (slower) get_body() function instead of get_sun()'s polynomial
+            approximation.
+        recompute : `bool`
+            If the column already exists, recompute it and overwrite.
+        """
+        if "sun_pos" in self.pointings.columns and not recompute:
+            return
+
+        # Compute and save the (RA, dec, dist) coordinates.
+        times = Time(self.pointings["obstime"], format="mjd")
+        if precise:
+            sun_pos = get_body("sun", times)
+        else:
+            sun_pos = get_sun(times)
+        self.pointings["sun_pos"] = sun_pos
+
+        # Compute and save the geocentric cartesian coordinates in AU.
+        (unit_x, unit_y, unit_z) = ang2unitvec(sun_pos.ra.degree, sun_pos.dec.degree)
+        self.pointings["sun_vec"] = np.array(
+            [
+                unit_x * sun_pos.distance.value,
+                unit_y * sun_pos.distance.value,
+                unit_z * sun_pos.distance.value,
+            ]
+        ).T
+
+    def append_unit_vector(self, recompute=False):
+        """Add the unit vector to the pointing data. Caches the result within the table
+        so that this computation only needs to be performed once.
+
+        Parameters
+        ----------
+        recompute : `bool`
+            If the column already exists, recompute it and overwrite.
+        """
+        if not "unit_vec" in self.pointings.columns or recompute:
+            (x, y, z) = ang2unitvec(self.pointings["ra"].data, self.pointings["dec"].data)
+            self.pointings["unit_vec"] = np.array([x, y, z]).T
+
+    def angular_dist_3d_heliocentric(self, cart_pt):
+        """Compute the angular offset (in degrees) between the pointing and
+        the 3-d location (specified as heliocentric coordinates and AU) of a point.
+
+        Parameters
+        ----------
+        cart_pt : tuple, list, or array
+            The point represented as (x, y, z) on which to compute the distances.
+
+        Returns
+        -------
+        ang_dist : numpy array
+            A length K numpy array with the angular distances in degrees.
+
+        Raises
+        ------
+        ValueError if the list of points is not width=3.
+        """
+        if len(cart_pt) != 3:
+            raise ValueError(f"Expected 3 dimensions, found {len(cart_pt)}")
+
+        if "sun_vec" not in self.pointings.columns:
+            self.append_sun_pos()
+        if "unit_vec" not in self.pointings.columns:
+            self.append_unit_vector(self)
```

Comment:
another small thing, but is there a reason why you're passing self here to a function called from self?